relayer/relays/beacon/header/syncer/syncer.go

```diff
@@ -447,6 +448,121 @@ func (s *Syncer) GetHeaderUpdate(blockRoot common.Hash, checkpoint *cache.Proof)
 	}, nil
 }
 
+func (s *Syncer) getBeaconStateAtSlot(slot uint64) (state.BeaconState, error) {
+	var beaconState state.BeaconState
+	beaconData, err := s.Client.GetBeaconState(strconv.FormatUint(slot, 10))
+	if err != nil {
+		return beaconState, fmt.Errorf("fetch beacon state: %w", err)
+	}
+
+	isDeneb := s.DenebForked(slot)
+
+	if isDeneb {
+		beaconState = &state.BeaconStateDenebMainnet{}
+	} else {
+		beaconState = &state.BeaconStateCapellaMainnet{}
+	}
+
+	err = beaconState.UnmarshalSSZ(beaconData)
+	if err != nil {
+		return beaconState, fmt.Errorf("unmarshal beacon state: %w", err)
+	}
+
+	return beaconState, nil
+}
+
+func (s *Syncer) GetFinalizedUpdateAtAttestedSlot(attestedSlot uint64) (scale.Update, error) {
+	var update scale.Update
+
+	// Get the beacon data first since it is mostly likely to fail
+	beaconState, err := s.getBeaconStateAtSlot(attestedSlot)
+	if err != nil {
+		return update, fmt.Errorf("fetch beacon state: %w", err)
+	}
+
+	finalizedCheckpoint := beaconState.GetFinalizedCheckpoint()
+
+	// Get the finalized header at the given slot state
+	finalizedHeader, err := s.Client.GetHeader(common.BytesToHash(finalizedCheckpoint.Root))
+	if err != nil {
+		return update, fmt.Errorf("fetch block: %w", err)
+	}
+
+	// Finalized header proof
+	stateTree, err := beaconState.GetTree()
+	if err != nil {
+		return update, fmt.Errorf("get state tree: %w", err)
+	}
+	_ = stateTree.Hash() // necessary to populate the proof tree values
+	finalizedHeaderProof, err := stateTree.Prove(FinalizedCheckpointGeneralizedIndex)
```

Comment:
Cool! A silly question how to quickly calculate the `GeneralizedIndex` for a specific field in beacon state, like the `FinalizedCheckpointGeneralizedIndex` 105 here?

Reply:
I think it can be calculated using the beacon state tree, but since it is well-known and won't change it is usually hardcoded, like here in the Ethereum spec: https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/light-client/sync-protocol.md#constants